src/main/java/programminglife/model/drawing/SubGraph.java

```diff
@@ -64,10 +64,23 @@ public SubGraph(DrawableSegment centerNode, int radius) {
         if (!this.nodes.containsKey(centerNode.getIdentifier())) {
             this.nodes.put(centerNode.getIdentifier(), centerNode);
         }
+    }
 
-        long genomeTime = System.nanoTime();
-        this.calculateGenomes();
-        Console.println("Time to find genomes through edge: " + (System.nanoTime() - genomeTime) / 1000000);
+    /**
+     * Detect SNPs and replace them.
+     */
+    public void replaceSNPs() {
+        Map<Integer, DrawableNode> nodesCopy = new LinkedHashMap<>(this.nodes);
+        for (Map.Entry<Integer, DrawableNode> entry : nodesCopy.entrySet()) {
```

Comment:
Nope. We want to remove elements in the collection, but not the current one. Using an `Iterator` still yields a `ConcurrentModificationException`.